docs/aws-sts-provisioning.md

```diff
@@ -0,0 +1,74 @@
+
+At present Hive does not automate the STS setup, rather we assume the user configures STS components manually and provides information to Hive. The following instructions refer to the 'ccoctl' tool coming in OpenShift 4.8 with the Cloud Credential Operator. This tool is not yet released but development is underway [here](https://github.com/openshift/cloud-credential-operator) and can be compiled from source, and used to prepare for provisioning 4.7 clusters.
+
+## Extract Credentials Requests from Desired OpenShift Release Image
+
+These vary from release to release, so be sure to use the same release image here as you plan to use for your ClusterDeployment:
+
+```bash
+$ mkdir credrequests/
+$ oc adm release extract quay.io/openshift-release-dev/ocp-release:4.7.1-x86_64 --credentials-requests --cloud=aws > credrequests/credrequests.yaml
+```
+
+
+## Setup STS Infrastructure
+
+```bash
+$ ccoctl create key-pair
+$ ccoctl create identity-provider --name-prefix mystsprefix --public-key-file serviceaccount-signer.public --region us-east-1
+$ ccoctl create iam-roles --credentials-requests-dir credrequests/ --identity-provider-arn <IdentityProviderARN> --name-prefix mystsprefix --region us-east-1
+```
+
+## Create Credentials Secret Manifests
+
+Hive allows passing in arbitrary Kubernetes resource manifests to pass through to the install process. We will leverage this to inject the Secrets and configuration required for an STS cluster.
+
+ccoctl will soon have a command to generate these Secrets but this is not implemented yet. For now you can follow the [manual STS documentation](https://docs.openshift.com/container-platform/4.7/authentication/managing_cloud_provider_credentials/cco-mode-sts.html) for how to create each Secret manually for the CredentialsRequests in your release image.
+
+Use the Role ARN's printed by `ccoctl create iam-roles` and the relevant target Secret namespace/name for each CredentialsRequest you extracted.
+
+Example:
+
+```yaml
+apiVersion: v1
+stringData:
+  credentials: |-
+    [default]
+    role_arn = arn:aws:iam::125931421481:role/mystsprefix-openshift-image-registry-installer-cloud-credentials
+    web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
+kind: Secret
+metadata:
+  name: installer-cloud-credentials
+  namespace: openshift-image-registry
+type: Opaque
+```
+
+
+### Create Authentication Manifest
+
+In the same directory as your Secret manifets, add another to configure the OpenShift Authentication operator to use the S3 OIDC provider created by `ccoctl create identity-provider`.
+
```

Comment:
NOTE:  the file name must be `cluster-authentication-02-config.yaml` in the manifests dir as https://github.com/openshift/installer/blob/82f88102baa6960cf64a59d8ed4a6d8a996839ff/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L169